packages/react/src/components/OutsideClick/index.tsx

```diff
@@ -0,0 +1,24 @@
+import { ComponentProps, ElementType, ReactNode } from 'react';
+import { useOnClickOutside } from '../../hooks/useOnClickOutside';
+
+type OutsideClickProp<Tag extends ElementType> = ComponentProps<Tag> & {
+  children: ReactNode;
+} & {
+  as?: Tag;
+  callback: () => void;
+};
+
+export function OutsideClick<
+  E extends HTMLElement = HTMLDivElement,
+  Tag extends ElementType = 'div'
+>({ as, children, callback, ...props }: OutsideClickProp<Tag>) {
+  const { ref } = useOnClickOutside<E>(callback);
+
+  const Component = as || 'div';
+
+  return (
+    <Component ref={ref} {...props}>
+      {children}
+    </Component>
+  );
+}
```

Comment:
다른 함수들과의 컨벤션을 맞추기위해 화살표 함수로 변경해주시면 감사드립니다! 

```suggestion
export const OutsideClick = <
  E extends HTMLElement = HTMLDivElement,
  Tag extends ElementType = "div"
>({
  as,
  children,
  callback,
  ...props
}: OutsideClickProp<Tag>) => {
  const { ref } = useOnClickOutside<E>(callback);

  const Component = as || "div";

  return (
    <Component ref={ref} {...props}>
      {children}
    </Component>
  );
};
```